core/sparse_array.h

```diff
@@ -57,12 +57,18 @@ class SparseArray
 	
 	uint32_t nrows_, ncols_;		// the number of rows and columns; determined at construction time
 	uint32_t nrows_set_;			// the number of rows that have been configured (at least partially, during building)
-	uint32_t nnz_;					// the number of non-zero entries in the sparse array (also at row_offsets[nrows_set])
 	uint32_t nnz_capacity_;			// the number of non-zero entries allocated for at present
 	
 	bool finished_;					// if true, Finished() has been called and the sparse array is ready to use
+	uint32_t nnz_;					// the number of non-zero entries in the sparse array (also at row_offsets[nrows_set])
 	
-	void _ResizeToFitNNZ(void);
+    //Stuff for multithreading
+    uint32_t initial_width;
+    uint32_t **columns;
+    sa_distance_t **distances;
+    uint32_t *nnz;
+    uint32_t *nnz_capacity;
```

Comment:
So this is precisely the discrepancy I was talking about that might be giving the Travis test error.  The old sparse array still exists because it is being used build _SA1() and SA3(). Technically, the remaining stuff is indeed "converting other parts of the code, outside of SparseArray, to use the new SparseArray design". I went ahead with changing only SA2() because that was the one targetted by the benchmark scripts and I wanted to be as less intrusive as possible to the original codebase for my first PR but I should have probably thought this through.